.github/workflows/dotnet-build-and-test.yml

```diff
@@ -20,57 +20,133 @@ permissions:
-          dotnet test -c ${{ matrix.configuration }} $project --no-build -v Normal --logger trx
-        done
-      env:
-        AzureOpenAI__Label: azure-text-davinci-003
-        AzureOpenAIEmbedding__Label: azure-text-embedding-ada-002
-        AzureOpenAI__DeploymentName: ${{ vars.AZUREOPENAI__DEPLOYMENTNAME }}
-        AzureOpenAIEmbeddings__DeploymentName: ${{ vars.AZUREOPENAIEMBEDDING__DEPLOYMENTNAME }}
-        AzureOpenAI__Endpoint: ${{ secrets.AZUREOPENAI__ENDPOINT }}
-        AzureOpenAIEmbeddings__Endpoint: ${{ secrets.AZUREOPENAI__ENDPOINT }}
-        AzureOpenAI__ApiKey: ${{ secrets.AZUREOPENAI__APIKEY }}
-        AzureOpenAIEmbeddings__ApiKey: ${{ secrets.AZUREOPENAI__APIKEY }}
-        Bing__ApiKey: ${{ secrets.BING__APIKEY }}
-        OpenAI__ApiKey: ${{ secrets.OPENAI__APIKEY }}
+      - uses: actions/checkout@v3
+
+      - name: Build dotnet solutions
+        run: |
+          export SOLUTIONS=$(find ./dotnet/ -type f -name "*.sln" | tr '\n' ' ')
+          for solution in $SOLUTIONS; do
+            dotnet build -c ${{ matrix.configuration }} /warnaserror $solution
+          done
+
+      - name: Run Unit Tests
+        run: |
+          export UT_PROJECTS=$(find ./dotnet -type f -name "*.UnitTests.csproj" | tr '\n' ' ')
+          for project in $UT_PROJECTS; do
+            dotnet test -c ${{ matrix.configuration }} $project --no-build -v Normal --logger trx
+          done
+
+      - name: Run Integration Tests
+        if: github.event_name != 'pull_request' && matrix.integration-tests
+        run: |
+          export INTEGRATION_TEST_PROJECTS=$(find ./dotnet -type f -name "*IntegrationTests.csproj" | tr '\n' ' ')
+          for project in $INTEGRATION_TEST_PROJECTS; do
+            dotnet test -c ${{ matrix.configuration }} $project --no-build -v Normal --logger trx
+          done
+        env:
+          AzureOpenAI__Label: azure-text-davinci-003
+          AzureOpenAIEmbedding__Label: azure-text-embedding-ada-002
+          AzureOpenAI__DeploymentName: ${{ vars.AZUREOPENAI__DEPLOYMENTNAME }}
+          AzureOpenAIEmbeddings__DeploymentName: ${{ vars.AZUREOPENAIEMBEDDING__DEPLOYMENTNAME }}
+          AzureOpenAI__Endpoint: ${{ secrets.AZUREOPENAI__ENDPOINT }}
+          AzureOpenAIEmbeddings__Endpoint: ${{ secrets.AZUREOPENAI__ENDPOINT }}
+          AzureOpenAI__ApiKey: ${{ secrets.AZUREOPENAI__APIKEY }}
+          AzureOpenAIEmbeddings__ApiKey: ${{ secrets.AZUREOPENAI__APIKEY }}
+          Bing__ApiKey: ${{ secrets.BING__APIKEY }}
+          OpenAI__ApiKey: ${{ secrets.OPENAI__APIKEY }}
+
+  # This final job is required to satisfy the merge queue. It must only run (or succeed) if no tests failed
+  dotnet-build-and-test-check:
+    if: always()
+    runs-on: ubuntu-latest
+    needs: [dotnet-build-and-test]
+    steps:
+      - name: Get Date
+        shell: bash
+        run: |
+          echo "date=$(date +'%m/%d/%Y %H:%M:%S')" >> "$GITHUB_ENV"
+
+      - name: Run Type is Daily
```

Comment:
Are you planning to use these job steps at all? They seem to be copied from the python-integration-tests, but are not used here (for python, the date and run_type variables are used for pushing a notification to teams)

Reply:
Yeah not used, just left for debugging purposes. Later one would like to refactor this templated job so it could be shared.